controller/resource_roles.go

```diff
@@ -87,6 +89,38 @@ func (c *ResourceRolesController) ListAssignedByRoleName(ctx *app.ListAssignedBy
 	})
 }
 
+// AssignRole assigns a specific role for a resource, to one or more identities.
+func (c *ResourceRolesController) AssignRole(ctx *app.AssignRoleResourceRolesContext) error {
```

Comment:
Controller method looks much better now, it's simply dealing with unmarshalling the request parameters from the payload and passing them to the service layer. :)